bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/test/java/com/google/cloud/bigtable/mapreduce/validation/EndToEndIT.java

```diff
@@ -0,0 +1,429 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.mapreduce.validation;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.bigtable.emulator.v2.BigtableEmulatorRule;
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import java.io.IOException;
+import java.util.Arrays;
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.HBaseTestingUtility;
+import org.apache.hadoop.hbase.HColumnDescriptor;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.HTableDescriptor;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.Admin;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Table;
+import org.apache.hadoop.hbase.mapreduce.BigtableTableHashAccessor;
+import org.apache.hadoop.hbase.mapreduce.HashTable;
+import org.apache.hadoop.hbase.mapreduce.SyncTable.SyncMapper.Counter;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.mapreduce.Counters;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * end to end integration test for HashTable and SyncTableValidationOnlyJob from Bigtable to HBase
+ * and HBase to Bigtable
+ */
+public class EndToEndIT {
+
+  private static final Log LOG = LogFactory.getLog(EndToEndIT.class);
+
+  private static final HBaseTestingUtility HB_TEST_UTIL = new HBaseTestingUtility();
+
+  @Rule public final BigtableEmulatorRule bigtableEmulator = BigtableEmulatorRule.create();
+
+  // Clients that will be connected to the emulator
+  private Connection bigtableConn;
+
+  private static final long TEST_TS = System.currentTimeMillis();
+
+  // bigtable
+  private static final String BT_TEST_PROJECT = "testproject-" + TEST_TS;
+  private static final String BT_TEST_INSTANCE = "testinstance-" + TEST_TS;
+
+  @BeforeClass
+  public static void setUpBeforeClass() throws Exception {
+    HB_TEST_UTIL.startMiniCluster(1);
+    HB_TEST_UTIL.setJobWithoutMRCluster();
+  }
+
+  @AfterClass
+  public static void tearDownAfterClass() throws Exception {
+    HB_TEST_UTIL.shutdownMiniCluster();
```

Comment:
I think we have a rule for HBASE_TEST_UTIL as well.

Reply:
we use HB_TEST_UTIL within [EndToEndIt for import snapshot job](https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/test/java/com/google/cloud/bigtable/mapreduce/hbasesnapshots/EndToEndIT.java#L61) as well. let me know if this is needed, we may need to address both tests.